apps/src/code-studio/responsive.js

```diff
@@ -0,0 +1,15 @@
+import debounce from 'lodash/debounce';
+import { getStore } from './redux';
+import { setResponsiveSize } from './responsiveRedux';
+
+/**
+ * Listen for page resize and dispatch events to Redux when if cross a
+ * responsive breakpoint width.
```

Comment:
Note that layouts don't only re-render when we cross a breakpoint width.  Many of them resize to 100% of current width, so they need to re-render whenever the window changes size (though debouncing helps reduce intensity).

Reply:
Can this happen with CSS instead of absolute sizing?

Reply:
Likewise with the course cards I'm testing this out with: can these have `float: left;` and let CSS sort out when to reflow them to the next line, based on the available width?

Maybe I'm not understanding how the React components are laid out on the page.

Reply:
As we discussed elsewhere, I retract my initial statement for now.  We do have pages whose layout changes in various ways between thresholds, but all of that is CSS rules being applied (often `width: 100%` in particular) rather than DOM changes occurring.  So, as you've pursued, it does make sense as an optimisation to only force the re-render (through changing state or otherwise) when we cross a threshold.  That said, we've survived with a debounced re-render of the DOM upon all resizes in /learn for the past year without known complaint.

As you pointed out, if we did start writing code that depended on values like current window width, then we'd want to re-render the DOM more often than just when crossing the threshold.  But afaik, we haven't done that yet.